kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java

```diff
@@ -60,8 +60,9 @@
     protected AtomicBoolean isActive = new AtomicBoolean(false);
     // Queue to make response get responseFuture in order and limit the max request size
     private final LinkedBlockingQueue<ResponseAndRequest> requestQueue;
-
-    protected final RequestStats requestStats;
+    @Getter
+    @Setter
+    protected RequestStats requestStats;
```

Comment:
Should we use a `Map` here?

Reply:
there is no need (maybe I can make this 'volatile')
because one KafkaRequestHandler serves only one tenant:

- before auth -> default RequestStats
- after auth -> the RequestStats for the logged tenant

Reply:
Got it. And yeah, it should be `volatile`.

Reply:
@BewareMyPower @Demogorgon314 I have updated the patch
PTAL when you have time